src/main/java/ca/wolfram/beta/symath/MathUtils.java

```diff
@@ -19,4 +24,31 @@ public static void print(String s, Object... o) {
         System.out.println(String.format(Locale.CANADA, s, o));
     }
 
+    public static Set<MathNode> getVariableTerms(MathNode node) {
+        Set<MathNode> set;
+        switch (node.getType()) {
+        case CONSTANT:
+        case MATH_CONSTANT:
+            set = new HashSet<MathNode>();
+            break;
+        case VARIABLE:
+            set = new HashSet<MathNode>();
+            set.add(node);
+            break;
+        default:
+            set = node.getChildren().stream().filter(children -> !children.isConstant()).collect(Collectors.toSet());
```

Comment:
Is this really what we want though? What if some terms are a division rather than multiplication? What if there is x ^ 2 and you ignore the 2? We should probably also verify negation somewhere